rooms.js

```diff
@@ -932,17 +932,25 @@ class BattleRoom extends Room {
 		this.resetUser = '';
 	}
 	requestKickInactive(user, force) {
-		if (this.resetTimer) {
-			if (user) this.sendUser(user, '|inactive|The inactivity timer is already counting down.');
+		if (this.resetTimer && user && (this.resetUser === user.userid || (user in this.game.players && this.resetUser === '+') || this.resetUser === '~')) {
```

Comment:
It will, but your new code will crash:

```
let userSetTimer = this.resetUser === user.userid || (user in this.game.players && this.resetUser === '+');
if (this.resetTimer && user && (userSetTimer || this.resetUser === '~'))
```

This reads `user.userid` before `user`.